framework/src/utils/MooseEnumBase.C

```diff
@@ -186,7 +186,7 @@ void
 MooseEnumBase::removeEnumerationName(std::string name)
 {
   std::transform(name.begin(), name.end(), name.begin(), ::toupper);
-  std::vector<std::string>::const_iterator iter = std::find(_names.begin(), _names.end(), name);
+  std::vector<std::string>::iterator iter = std::find(_names.begin(), _names.end(), name);
```

Comment:
This is a compile error on Intel? Wow! Have you tried catching it as a `const auto` instead?